include/ht_file.h

```diff
@@ -19,8 +19,6 @@
 #include <ht_file_interface.h> //IFile
 #include <string> //std::string typedef
 #include <cstddef> //size_t typedef
```

Comment:
Well, if IFile decides for some reason that it doesn't need std::fstream, it can't remove its include anymore because of the downward stream of files that rely on it to get access to fstream.  We're trying to avoid that and allow IFile to include / not include whatever it needs, and not have to worry about what happens downstream.